### PR TITLE
Add FhirPath math for date(Time)/duration additions.

### PR DESCRIPTION
**Pull request by ewoutkramer**
Also applied changes pertaining to https://jira.hl7.org/browse/FHIR-28144 and the discussion in https://chat.fhir.org/#narrow/stream/179266-fhirpath/topic/Date.2FTime.20comparison.20vs.20equality

### CODE REVIEW

**Comment by ewoutkramer** [comment]
This is basically PR #164, plus the changes to equality/comparisons due to  https://jira.hl7.org/browse/FHIR-28144 and the discussion in https://chat.fhir.org/#narrow/stream/179266-fhirpath/topic/Date.2FTime.20comparison.20vs.20equality

File: src/Hl7.FhirPath.Tests/Tests/BasicFunctionTests.cs [hunk 1]
```diff
@@ -189,7 +189,7 @@ public void TestExpressionTodayFunction()
             Assert.AreEqual(P.Date.Today(), scalar("today()"));
 
             // Check greater than
-            isB("today() < @" + P.Date.FromDateTimeOffset(DateTimeOffset.UtcNow.AddDays(1), includeOffset: false));
+            isB("today() < @" + P.Date.FromDateTimeOffset(DateTimeOffset.UtcNow.AddDays(2), includeOffset: false));
```
Comment (marcovisserFurore) [inline src/Hl7.FhirPath.Tests/Tests/BasicFunctionTests.cs:192]: Why is this changed to +2 days instead of 1?

**Review by marcovisserFurore — CHANGES_REQUESTED** [review]
See my remark

File: src/Hl7.FhirPath.Tests/Tests/BasicFunctionTests.cs (hunk 1 above)
Comment (ewoutkramer) [reply src/Hl7.FhirPath.Tests/Tests/BasicFunctionTests.cs:192]: This test will otherwise intermittently fail depending on the time of day:  if you are east of GMT and it is already the next day where you are (compared to GMT), this means `today()` is already UTC + 1 day and is not `<`.  Doing `today() <= @.....` would also fix it.

Comment (ewoutkramer) [reply src/Hl7.FhirPath.Tests/Tests/BasicFunctionTests.cs:192]: (Brian found this bug, who is at GMT+10, so he runs into this much more often than we do)

**Review by marcovisserFurore — APPROVED** [review]
